apps/sms/test/marionette/drafts_test.js

```diff
@@ -27,95 +29,135 @@ marionette('Messages Drafts', function() {
     }
   });
 
-  var messagesApp, activityCallerApp, storage;
+  var messagesApp, storage;
 
-  function createAndSaveDraft(draft) {
+  function createAndSaveDraft(view, draft) {
     // Add text content.
     if (draft.text) {
-      messagesApp.Composer.messageInput.tap();
-      messagesApp.Composer.messageInput.sendKeys(draft.text);
+      view.typeMessage(draft.text);
     }
 
     // Add subject.
     if (draft.subject) {
-      messagesApp.showSubject();
-      messagesApp.Composer.subjectInput.sendKeys(draft.subject);
+      view.showSubject();
+      view.typeSubject(draft.subject);
     }
 
     // Add recipient.
     if (draft.recipients) {
       draft.recipients.forEach(function(recipient) {
-        messagesApp.addRecipient(recipient);
+        view.addNewRecipient(recipient);
       });
     }
 
     // Add attachment.
     if (draft.shouldHaveAttachment) {
-      client.waitFor(function() {
-        return messagesApp.Composer.attachButton.enabled();
-      });
-      messagesApp.Composer.attachButton.tap();
-      messagesApp.selectSystemMenuOption('Messages Activity Caller');
-
-      activityCallerApp.switchTo();
-      activityCallerApp.pickImage();
+      view.addAttachment().choose('Messages Activity Caller').pickImage();
+      messagesApp.switchTo();
     }
 
-    messagesApp.switchTo();
-    messagesApp.performHeaderAction();
+    var inboxView = view.backToInbox();
 
-    messagesApp.selectAppMenuOption('Replace existing Draft');
-  }
+    // It's important to wait for the updated thread node (when old node is
+    // removed, and new one is inserted), otherwise we can get old node
+    // reference and when we try to tap on it, we'll get stale element
+    // reference exception if node has been removed in between.
+    var conversationWithDraft = null;
+    client.waitFor(function() {
+      var conversations = inboxView.conversations;
+
+      if (!conversations.length) {
+        return false;
+      }
 
-  function assertIsFocused(element, message) {
-    assert.isTrue(element.scriptWith(function(el) {
-      return document.activeElement === el;
-    }), message);
+      conversationWithDraft = conversations[0];
+
+      if (!conversationWithDraft.isDraft && !conversationWithDraft.hasDraft) {
+        return false;
+      }
+
+      return conversationWithDraft.bodyText ===
+        (draft.subject || draft.shouldHaveAttachment ? '' : draft.text);
+    });
+
+    return conversationWithDraft;
   }
 
-  function assertDraft(draft) {
-    var conversation = messagesApp.Inbox.firstConversation;
-    assert.ok(conversation.getAttribute('class').indexOf('draft') !== -1);
+  function discardDraft(view, draft) {
+    if (draft.text || draft.shouldHaveAttachment) {
+      view.clearMessage();
+    }
+
+    if (draft.subject) {
+      view.hideSubject();
+    }
+
+    if (draft.recipients) {
+      view.clearRecipients();
+    }
```

Comment:
> also, does view.clearRecipients work in a Conversation ?

Nope, we don't have such method in ConversationView and that's why I have ```if``` conditions here :)